### PR TITLE
Make libgcc_s relocatable on Macos by adding @rpath to installname

### PR DESCRIPTION
**Pull request by gartung**
This changes the installname of libgcc_s.1.dylib on OSX from an absolute path set by --prefix to @rpath/libgcc_s.1.dylib. This is important for relocating libgcc_s.1.dylib because install_name_tool will not change the ID but with add/change rpath. Has no impact on linux.

### CODE REVIEW

**Comment by gartung** [comment]
Looks like I need to have gcc add its own rpath first.

**Comment by gartung** [comment]
@tgamblin Nevermind the previous comment. I see that the gcc executables don't use libgcc_s.

**Comment by gartung** [comment]
@eschnett Is this of interest?

**Comment by gartung** [comment]
I quit!

**Comment by gartung** [comment]
Try again

File: var/spack/repos/builtin/packages/gcc/package.py [hunk 1]
```diff
@@ -147,6 +151,11 @@ def write_rpath_specs(self):
             for line in lines:
                 out.write(line + "\n")
                 if line.startswith("*link:"):
-                    out.write("-rpath %s/lib:%s/lib64 \\\n" %
-                              (self.prefix, self.prefix))
+                    if sys.platform == 'darwin':
+                        out.write("-rpath %s/lib -rpath %s/lib64 "
+                                  "-headerpad_max_install_names \\\n" %
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gcc/package.py]: If you use r"string" instead of "string" you won't need to escape your `\`.

**Comment by adamjstewart** [comment]
@davydden You use macOS, can you review this?

**Comment by gartung** [comment]
@adamjstewart please revert. This hasn't been tested with ElCapitan or Sierra.

**Pull request merged**